spec/models/provider_spec.rb

```diff
@@ -34,38 +34,146 @@
+          it 'returns json_data from the first enrichment' do
+            enrichment = build(:provider_enrichment)
+            provider = create(:provider, enrichments: [enrichment])
+
+            expect(provider.address_info).to eq(
+              'address1' => enrichment.address1,
+              'address2' => enrichment.address2,
+              'address3' => enrichment.address3,
+              'address4' => enrichment.address4,
+              'postcode' => enrichment.postcode,
+              'region_code' => ProviderEnrichment.region_codes[enrichment.region_code]
+            )
+          end
+
+          it 'returns json_data from the newest enrichment' do
+            enrichment = build(:provider_enrichment)
+            newest_enrichment = build(:provider_enrichment, created_at: Date.today)
+            provider = create(:provider, enrichments: [enrichment, newest_enrichment])
+
+            expect(provider.address_info).to eq(
+              'address1' => newest_enrichment.address1,
+              'address2' => newest_enrichment.address2,
+              'address3' => newest_enrichment.address3,
+              'address4' => newest_enrichment.address4,
+              'postcode' => newest_enrichment.postcode,
+              'region_code' => ProviderEnrichment.region_codes[newest_enrichment.region_code]
+            )
+          end
+        end
+        context 'enrichment has partial set for address_info' do
+          it 'returns address of the enrichment' do
+            enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
+            'website' => Faker::Internet.url,
+            'address1' => Faker::Address.street_address,
+            'postcode' => nil,
+            'train_with_us' => Faker::Lorem.sentence.to_s,
+            'train_with_disability' => Faker::Lorem.sentence.to_s })
+            provider = create(:provider, enrichments: [enrichment])
+
+            expect(provider.address_info).to eq(
+              'address1' => enrichment.address1,
+              'address2' => enrichment.address2,
+              'address3' => enrichment.address3,
+              'address4' => enrichment.address4,
+              'postcode' => enrichment.postcode,
+              'region_code' => ProviderEnrichment.region_codes[enrichment.region_code]
+            )
+          end
+
+          context 'enrichment has only region code set for address_info' do
+            it 'returns address of the provider' do
+              london = ProviderEnrichment.region_codes['London']
+              enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
+              'website' => Faker::Internet.url,
+              'region_code' => 'No region',
+              'train_with_us' => Faker::Lorem.sentence.to_s,
+              'train_with_disability' => Faker::Lorem.sentence.to_s })
+              provider = create(:provider, region_code: london, enrichments: [enrichment])
```

Comment:
We need to remove all the address-related fields other than `region_code` from the actual JSON blob to properly simulate what `manage-courses-api` stuffs into the DB. So use this bit of code from above:

```ruby
            # forcing it to be absent in json_data altogether
            ProviderEnrichment.connection.update(<<~EOSQL)
              UPDATE provider_enrichment
                    SET json_data=json_data-'Address1'-'Address2'-'Address3'-'Address4'-'Postcode'-'RegionCode'
                    WHERE provider_code='#{enrichment.id}'
            EOSQL
```